viewer/packages/metrics/src/MetricsLogger.ts

```diff
@@ -115,6 +120,56 @@ export class MetricsLogger {
     MetricsLogger.trackEvent('cadModelStyleAssigned', { nodeCollectionClassToken, style: appearance });
   }
 
+  private static readonly trackCadNodeTransformOverriddenVars = {
+    zeroVector: new THREE.Vector3(0, 0, 0),
+    oneVector: new THREE.Vector3(1, 1, 1),
+    identityRotation: new THREE.Quaternion().identity(),
+
+    translation: new THREE.Vector3(),
+    scale: new THREE.Vector3(),
+    rotation: new THREE.Quaternion()
+  };
+
+  /**
+   * Track use of CAD node transform overrides. Note that the metric is throttled and will only trigger
+   * once per second.
+   * @param nodeCount Number of nodes affected by the transform override
+   * @param matrix  Matrix used to override the node transform
+   */
+  static readonly trackCadNodeTransformOverridden = throttle(
+    (nodeCount: number, matrix: THREE.Matrix4) => MetricsLogger.trackCadNodeTransformOverriddenImpl(nodeCount, matrix),
+    MetricsLogger.TrackCadNodeTransformOverriddenThrottleDelay
+  );
+
+  private static trackCadNodeTransformOverriddenImpl(nodeCount: number, matrix: THREE.Matrix4): void {
+    const { zeroVector, oneVector, identityRotation, translation, scale, rotation } =
+      MetricsLogger.trackCadNodeTransformOverriddenVars;
+    matrix.decompose(translation, rotation, scale);
+
+    const hasRotation = Math.abs(rotation.dot(identityRotation) - 1.0) > 1e-5; // Angle between two quaternions is arccos(q1.dot(q2))
```

Comment:
I don't think this formula is right. I believe we want what's described here: https://se.mathworks.com/matlabcentral/answers/415936-angle-between-2-quaternions . In essence, I believe the formula should be

$$\theta = 2 cos^{-1}((q_1 \times q_2^{-1}).w)$$
Where the cross denotes quaternion multiplication

Reply:
Fixed.

Reply:
Okay, I guess it's good enough :P